doc/crypto/api/ops/ka.rst

```diff
@@ -9,11 +9,19 @@
 Key agreement
 =============
 
```

Comment:
We now use “raw key agreement” to mean three different things:

* An algorithm like `PSA_ALG_ECDH`, as opposed to `PSA_ALG_KEY_AGREEMENT(..., ...)`
* `psa_raw_key_agreement` or `psa_key_agreement`, as opposed to `psa_key_derivation_key_agreement`
* `psa_raw_key_agreement`, as opposed to `psa_key_agreement` and `psa_key_derivation_key_agreement`

I wonder if we should revise this. The only one we can change is the name of the function `psa_raw_key_agreement`. Unfortunately, the first two are the ones that make the more sense and that are closely related (a raw algorithm is what you pass to the “raw” functions, i.e. the functions that only do a key agreement and nothing else). Should we switch to a different adjective for “key agreement not combined with a key derivation”? Maybe “standalone”?

Reply:
I have already used 'standalone key agreement' in the section title for `psa_key_agreement()` and `psa_raw_key_agreement()`. Elsewhere we already use 'combined key agreement' to describe the ka+kdf algorithms.

I think it makes sense to adjust terminology throughout the spec on this topic. There are a few other API instances, that we might want to reconsider:

* `PSA_ALG_IS_RAW_KEY_AGREEMENT()`
* `PSA_RAW_KEY_AGREEMENT_OUTPUT_SIZE()`
* `PSA_RAW_KEY_AGREEMENT_OUTPUT_MAX_SIZE`

Given that `psa_key_derivation_key_agreement()` has no 'output', we could add synonyms for these three macros, for example:

* `PSA_ALG_IS_STANDALONE_KEY_AGREEMENT()`
* `PSA_KEY_AGREEMENT_OUTPUT_SIZE()`
* `PSA_KEY_AGREEMENT_OUTPUT_MAX_SIZE`

Reply:
I've addressed the easy aspects in https://github.com/ARM-software/psa-api/pull/101/commits/ed15b615064c16339b3c53be22a6361eae1b2967, but would appreciate some discussion on whether we should also do something with the interface elements.

Reply:
Let's add `PSA_ALG_IS_STANDALONE_KEY_AGREEMENT()` as a synonym.

But we don't need to change buffer size macros since they aren't needed for `psa_key_agreement`, only to determine output buffer sizes for `psa_raw_key_agreement`.

Reply:
Added https://github.com/ARM-software/psa-api/pull/101/commits/b4869518646ddbe2af518286d2a160c8c4eb6212